examples/ecs/component_hooks.rs

```diff
@@ -1,4 +1,11 @@
-//! This examples illustrates the different ways you can employ component lifecycle hooks
+//! This example illustrates the different ways you can employ component lifecycle hooks.
+//!
+//! Whenever possible, prefer using Bevy's Events for reacting to component changes.
+//! Events generally offer better performance and more flexible integration into Bevy's systems.
+//! Hooks are useful in specific scenarios but have limitations (only one hook per component,
+//! less ergonomic than events).
+//!
+//! In the future, Bevy's Observer features might address some of these limitations.
```

Comment:
Long term plans like these that may take a significant amount of time to materialize and get merged shouldn't be in a user-facing example.